### PR TITLE
Add support of schema_info for AsdfSearchResult as "path"

### LINKED ISSUE #558: Provide a way of getting schema information given a node

### PR DESCRIPTION
**Pull request by WilliamJamieson**
Adds support for using a `AsdfSearchResult` as the `path` argument to a `schema_info` call on `AsdfFile` and a `schema_info` method on an `AsdfSearchResult`.

Fixes #558

### CODE REVIEW

File: asdf/_node_info.py [hunk 1]
```diff
@@ -5,7 +5,54 @@
 from .treeutil import get_children
 
 
-def collect_schema_info(key, path, node, identifier="root", preserve_list=True, refresh_extension_manager=False):
+def _filter_tree(info, filters):
+    """
+    Remove nodes from the tree that get caught in the filters.
```
Comment (perrygreenfield) [inline asdf/_node_info.py:10]: I find this description confusing. Shouldn't say something more like keep nodes in the tree that pass the filters? Getting caught in a filter is not as clear to me.

**Review by perrygreenfield — COMMENTED** [review]
I'm somewhat uncertain about how multiple search results are handled.

File: docs/asdf/features.rst [hunk 2]
```diff
@@ -754,12 +754,26 @@ the information from the schema requested together with the value stored in the
...
 
     >>> af.schema_info("archive_catalog", "foo.bar")  # doctest: +SKIP
     {'thing1': {'archive_catalog': 'Thing 1 Archive catalog information'},
      'thing2': {'archive_catalog': 'Thing 2 Archive catalog information'}}
+
+Or one can provide a ``path`` as an `asdf.search.AsdfSearchResult` object:
+
+.. code:: pycon
+
+    >>> af.schema_info("archive_catalog", af.search("bar"))  # doctest: +SKIP
```
[the remark this answers is not in the document]
Comment (WilliamJamieson) [reply docs/asdf/features.rst:771]: `af.search` returns an `AsdfSearchResult` object, which we can now parse with `schema_info`.

**Review by perrygreenfield — APPROVED** [review]
LGTM

**Review by nden — APPROVED** [review]

**Pull request merged**